packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -187,14 +187,8 @@ describe('<DataGrid /> - Filter', () => {
     });
   });
 
-  describe('Date operators', function test() {
-    const isEdge = /Edg/.test(window.navigator.userAgent);
-    before(function before() {
-      if (isEdge) {
-        // We need to skip edge as it does not handle the date the same way as other browsers.
-        this.skip();
-      }
-    });
+  describe('date operators', () => {
+    const formatter = new Intl.DateTimeFormat('en');
```

Comment:
should we use this in the date column definition 🤔 ?

Reply:
I changed to `params.value.toLocaleDateString('en-US')` because it's simpler to use. 

I couldn't find any difference between the two implementations, they even accept the same arguments. In a small performance test I did, `toLocaleDateString` ran slightly faster, so I would keep it. https://jsbench.me/l2kowy4klg/3

Reply:
why not integrate the locale detection in the date col def format function, and do the same for numbers?

Reply:
> why not

Because we want to use the default locale by default for end-users and the same locale for all the test environments?

Reply:
How about we use the locale used in the translation to actually format the dates and numbers?

In tests, we will have to either override the date formatter to force a pre-defined locale, or override the locale detector...
but yes in tests we would need a static locale.

Reply:
> How about we use the locale used in the translation to actually format the dates and numbers?

We talk about this problem in https://github.com/mui-org/material-ui/issues/24495. It's the same problem for the formatting of numbers.